### ISSUE TITLE
makedep.py fails when non-ASCII characters are present

### ISSUE THREAD

**Issue by edditler** [cp2k/cp2k#976]
If one of the source files contains characters such as "∂" (in a comment) `make test` will fail:

```
    parse_file(parsed_files, fn, src_dir)  # parses also included files
  File "./tools/build_utils/makedep.py", line 204, in parse_file
    content = fhandle.read()
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xf0 in position 22944: ordinal not in range(128)
Makefile:416: recipe for target 'makedep' failed
make[3]: *** [makedep] Error 1
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
ERROR: make -j 1 ARCH=local  VERSION=sopt failed

Summary: Compilation failed.
Status: FAILED
```

It seems to me that there is no harm in allowing arbitrary characters in the comments if the compilers can handle them. Also, using UTF-8 as the encoding should not affect the ASCII symbols.

Works fine: `make -j 1 ARCH=local VERSION=sopt`
Fails: `make test`

**Comment by dev-zero** [comment]
I agree, especially now that we're using python 3 exclusively which is unicode native by default. It should be enough to drop the following line:

https://github.com/cp2k/cp2k/blob/3f90c5beae60370297fc2673d82acf41f18abdea/tools/regtesting/do_regtest#L13

or rather, replace `C` with `C.UTF-8` or `en_US.UTF-8` instead, since we still want any generated error messages to be in english for when people are reporting them.

**Comment by dev-zero** [comment]
... and `tools/conventions/analyze_src.py` also needs to be nerfed.

**Comment by edditler** [comment]
👍  should I create the pull request?

**Comment by oschuett** [comment]
Yes, please go ahead and prepare a suitable PR.

**Comment by edditler** [comment]
I hope I understood correctly that the check in tools/conventions/analyze_src.py is not necessary at all. https://github.com/cp2k/cp2k/pull/996

**Comment by oschuett** [comment]
>  if the compilers can handle them.

Maybe this is a bit late, but are we actually sure that all compilers can handle unicode comments?

**Comment by dev-zero** [comment]
> > if the compilers can handle them.
> 
> Maybe this is a bit late, but are we actually sure that all compilers can handle unicode comments?

UTF-8 unicode is specifically designed to not contain any ASCII characters (0-127) in a multibyte sequence. We are therefore safe if we stick to `\n` as a newline delimiter and UTF-8 encoding. Meaning that the compiler can safely treat everything after an unquoted `!` as a single character byte ignoring the encoding completely and be able to find the line termination.

**Comment by oschuett** [comment]
> the compiler can safely treat everything after an unquoted ! as a single character byte ignoring the encoding completely 

True, but are all compilers actually behaving this way?

**Comment by dev-zero** [comment]
resolved with a4c43d8a6bce64255d16f4797d18fbb03215d330
